src/common/mmap.h

```diff
@@ -124,6 +139,9 @@ struct map_tracker {
 #endif
 };
 
+extern os_rwlock_t Mmap_list_lock;
```

Comment:
This should never be needed. The interface can be self-contained. extern locks are almost always bad.
In this case all you need is a foreach callback as far as I can see.